Scemantic Focus

```diff
@@ -0,0 +1,110 @@
+---
+id: 
+name: Scemantic Focus 
+rule_type: atomic
+
+description: | 
+ This rule checks that every element part of the sequential focus navigation has a semantic role.
+
+accessibility_requirements: 
+  wcag20:2.4.4: # Link Purpose (In Context) 
```

Comment:
I don't believe this is a failure of 2.4.4. More generally, while I'm totally on board with this rule, I don't believe this is a failure of WCAG 2.1 at all. I've had that discussion in the past, this rule looks quite a lot like something we have in axe-core today. Here's a link to that conversation:
https://github.com/dequelabs/axe-core/issues/632

Reply:
Anna suggested 4.1.2

Reply:
> 
> 
> Anne suggested 4.1.2

This would also not fail 4.1.2

4.1.2 asks a role of all "user interface components". But if a piece of text with a span on it suddenly receives focus (because of tabindex for example) that doesn't make that piece of text a user interface component because it doesn't control anything.

I agree that it is a good rule (because most of the time it will point to a 4.1.2 violation) but it is currently not necessarily a WCAG failure to fail this rule.